pkg/auth/api/proto/auth.proto

```diff
@@ -0,0 +1,1170 @@
+    repeated string project_client_ids = 6;
+    string token_id = 7;
+    google.protobuf.Timestamp token_expiration = 8;
+    string user_id = 9;
+}
+
+message TokenID {
+    string id = 1;
+}
+
+message UserSessionID {
+    string id = 1;
+    string agent_id = 2;
+}
+
+message UserSessions {
+    repeated UserSession user_sessions = 1;
+}
+
+message UserSession {
+    string id = 1;
+    string agent_id = 2;
+    UserSessionState auth_state = 3;
+    AuthUser user = 4;
+    bool password_verified = 5;
+    MfaType mfa = 6;
+    bool mfa_verified = 7;
+    google.protobuf.Timestamp auth_time = 8;
+}
+
+message UserSessionViews {
+    repeated UserSessionView user_sessions = 1;
+}
+message UserSessionView {
+    string id = 1;
+    string agent_id = 2;
+    UserSessionState auth_state = 3;
+    string user_id = 4;
+    string user_name = 5;
+}
+
+message AuthUser {
+    string user_id = 1;
+    string user_name = 2;
+}
+
+enum UserSessionState {
+    USERSESSIONSTATE_UNSPECIFIED = 0;
+    USERSESSIONSTATE_ACTIVE = 1;
+    USERSESSIONSTATE_TERMINATED = 2;
+}
+
+message AuthSessionID {
+    string id = 1;
+    string agent_id = 2;
+    BrowserInformation browser_info = 3;
+}
+
+message SelectUserRequest {
+    string agent_id = 1;
+    string auth_session_id = 2;
+    string user_session_id = 3;
+    BrowserInformation browser_info = 4 [(validate.rules).message.required = true];
+}
+
+message VerifyUserRequest {
+    string agent_id = 1 [(validate.rules).string.min_len = 1];
+    string auth_session_id = 2 [(validate.rules).string.min_len = 1];
+    string user_name = 3 [(validate.rules).string = {min_len: 1, max_len: 200}];
+    BrowserInformation browser_info = 4 [(validate.rules).message.required = true];
+}
+
+message VerifyPasswordRequest {
+    string agent_id = 1 [(validate.rules).string.min_len = 1];
+    string auth_session_id = 2 [(validate.rules).string.min_len = 1];
+    string password = 3 [(validate.rules).string = {min_len: 1, max_len: 72}];
+    BrowserInformation browser_info = 4 [(validate.rules).message.required = true];
+}
+
+message VerifyMfaRequest {
+    string agent_id = 1 [(validate.rules).string.min_len = 1];
+    string auth_session_id = 2 [(validate.rules).string.min_len = 1];
+    BrowserInformation browser_info = 3;
+    oneof mfa {
+        AuthSessionMultiFactorOTP otp = 4;
+    }
+}
+
+message AuthSessionMultiFactorOTP {
+    string code = 1 [(validate.rules).string = {max_len: 200}];
+}
+
+message NextStep {
+    NextStepType type = 1;
+    oneof data {
+        LoginData login = 2;
+        PasswordData password = 3;
+        MfaVerifyData mfa_verify = 4;
+        MfaPromptData mfa_prompt = 5;
+        ChooseUserData choose_user = 6;
+    }
+}
+
+enum NextStepType {
+    NEXTSTEP_UNSPECIFIED = 0;
+    NEXTSTEP_LOGIN = 1;
+    NEXTSTEP_PASSWORD = 2;
+    NEXTSTEP_CHANGE_PASSWORD = 3;
+    NEXTSTEP_MFA_PROMPT = 4; //TODO: ?
+    NEXTSTEP_MFA_INIT_CHOICE = 5; //TODO: ?
+    NEXTSTEP_MFA_INIT_CREATE = 6;
```

Comment:
i believe we they're all "implemented" by step NEXTSTEP_MFA_INIT_CREATE... so we could delete the others

Reply:
i removed also     
NEXTSTEP_MFA_INIT_VERIFY = 7;
NEXTSTEP_MFA_INIT_DONE = 8;